account_usability/account.py

```diff
@@ -179,15 +179,14 @@ def _compute_sales_dates(self):
             returned string: "sale1 (date1), sale2 (date2) ..."
         """
         for inv in self:
-            sale_ids = [x.sale_line_ids.mapped('order_id')[0].id
-                        for x in inv.invoice_line_ids]
-            sales = self.env["sale.order"].browse(set(sale_ids))
+            sales = inv.invoice_line_ids.mapped(
+                'sale_line_ids').mapped('order_id')
             lang = inv.partner_id.commercial_partner_id.lang
             date_format = self.env["res.lang"]._lang_get(
                 lang or "").date_format
-            dates = ["%s%s" % (
-                     x.name, x.confirmation_date and
-                     " (%s)" % x.confirmation_date.strftime(date_format) or "")
+            dates = ["%s%s" % (x.name, x.confirmation_date and
+                               " (%s)" %
+                               x.confirmation_date.strftime(date_format) or "")
```

Comment:
["%s (%s)" % (x.name, x.confirmation_date.strftime(date_format) for x in sales if sale.confirmation_date]

Reply:
and if sale but no confirmation_date, you missed a date.
you can't be sure than a day the code'll not change and you can fall in this case.